### PR TITLE
feat: Add valid config fields to its help

### PR DESCRIPTION
**Pull request by dbolson**
Running `ldcli config --help` shows
```
View and modify specific configuration values

Current settings:
- `access-token`: LaunchDarkly access token with write-level access
- `analytics-opt-out`: Opt out of analytics tracking
- `base-uri`: LaunchDarkly base URI
- `output`: Command response output format in either JSON or plain text

Usage:
  ldcli config [flags]

Flags:
  -h, --help           help for config
      --list           List configs
      --set            Set a config field to a value
      --unset string   Unset a config field

Global Flags:
      --access-token string   LaunchDarkly access token with write-level access
      --analytics-opt-out     Opt out of analytics tracking
      --base-uri string       LaunchDarkly base URI (default "https://app.launchdarkly.com")
  -o, --output string         Command response output format in either JSON or plain text (default "plaintext")`
```

**Requirements**

- [ ] I have added test coverage for new or changed functionality
- [ ] I have followed the repository's [pull request submission guidelines](../blob/main/CONTRIBUTING.md#submitting-pull-requests)
- [ ] I have validated my changes against all supported platform versions

**Related issues**

Provide links to any issues in this repository or elsewhere relating to this pull request.

**Describe the solution you've provided**

Provide a clear and concise description of what you expect to happen.

**Describe alternatives you've considered**

Provide a clear and concise description of any alternative solutions or features you've considered.

**Additional context**

Add any other context about the pull request here.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -32,11 +32,12 @@ ldcli --help
 
 The LaunchDarkly CLI allows you to save preferred settings, either within a config file using the `config` commands, or set as environment variables.
 
-Current respected settings:
+Current settings:
```
Comment (dbolson) [inline README.md:35]: Updated the readme to be consistent with the config help output.

File: cmd/config/config.go [hunk 2]
```diff
@@ -39,6 +40,22 @@ func NewConfigCmd(analyticsTracker analytics.Tracker) *cobra.Command {
 		},
 	}
 
+	helpFun := cmd.HelpFunc()
+	cmd.SetHelpFunc(func(cmd *cobra.Command, args []string) {
```
Comment (dbolson) [inline cmd/config/config.go:44]: This runs the existing help fn after updating the command's `Long` value.

File: cmd/cliflags/flags.go [hunk 3]
```diff
@@ -13,4 +13,18 @@ const (
 	OutputFlag      = "output"
 	ProjectFlag     = "project"
 	RoleFlag        = "role"
+
+	AccessTokenFlagDescription = "LaunchDarkly access token with write-level access"
```
Comment (dbolson) [inline cmd/cliflags/flags.go:17]: Moved these to constants to remove duplication.

File: cmd/config/config.go [hunk 4]
```diff
@@ -39,6 +40,22 @@ func NewConfigCmd(analyticsTracker analytics.Tracker) *cobra.Command {
...
+		var sb strings.Builder
+		sb.WriteString("\n\nCurrent settings:\n")
+		for _, s := range []string{
```
Comment (dbolson) [inline cmd/config/config.go:47]: This uses a slice to preserve order since a map doesn't.

File: README.md (hunk 1 above)
Comment (bwoskow-ld) [reply README.md:35]: How about:

```suggestion
Supported settings:
```
?

File: README.md [hunk 5]
```diff
@@ -32,11 +32,12 @@ ldcli --help
...
 
-* `access-token` A LaunchDarkly API token with write-level access
+* `access-token` A LaunchDarkly access token with write-level access
 * `analytics-optout` Opt out of analytics tracking (default false)
```
Comment (bwoskow-ld) [inline README.md:38]:
```suggestion
* `analytics-opt-out` Opt out of analytics tracking (default false)
```

File: cmd/config/config.go [hunk 6]
```diff
@@ -39,6 +40,22 @@ func NewConfigCmd(analyticsTracker analytics.Tracker) *cobra.Command {
 		},
 	}
 
+	helpFun := cmd.HelpFunc()
+	cmd.SetHelpFunc(func(cmd *cobra.Command, args []string) {
+		var sb strings.Builder
+		sb.WriteString("\n\nCurrent settings:\n")
```
Comment (bwoskow-ld) [inline cmd/config/config.go:46]: How about:
```suggestion
		sb.WriteString("\n\nSupported settings:\n")
```
?

File: cmd/config/testdata/help.golden [hunk 7]
```diff
@@ -10,7 +16,7 @@ Flags:
       --unset string   Unset a config field
```
Comment (bwoskow-ld) [inline cmd/config/testdata/help.golden:16]: Minor (and I know, I'm tagging this on a test file) but I noticed that `--unset` is marked as taking a string parameter whereas `--set` isn't marked to have any parameter. It should also have `string`.

File: README.md (hunk 1 above)
Comment (dbolson) [reply README.md]: Updated

File: cmd/config/testdata/help.golden (hunk 7 above)
Comment (dbolson) [reply cmd/config/testdata/help.golden:16]: https://app.shortcut.com/launchdarkly/story/242359/config-set-should-say-it-takes-a-string

**Review by sunnyguduru — APPROVED** [review]

**Review by bwoskow-ld — APPROVED** [review]

**Pull request merged**